### PR TITLE
[COR-106] Add otel collector timeout

### PR DESCRIPTION
**Pull request by gunnarsundberg**
COR-106

Add timeout so tracing does not cause connectors to hang

### CODE REVIEW

**Comment by rukolahasser** [comment]
Is this PR still needed?

**Comment by gunnarsundberg** [comment]
@rukolahasser yes. there was a similar pr that was approved for streamserver, but this one is still awaiting a review

File: connector.go [hunk 1]
```diff
@@ -90,7 +91,9 @@ func NewConnector(options ...Option) (*Connector, error) {
 	log.Info().Str("addr", "0.0.0.0:8080").Msg("healthcheck listening on /live and /ready")
 
 	//	Initialize trace provider
-	_, err = monitor.InitTracerProvider(context.TODO(), conf.GetString("trace.host.grpc"), c.id(), c.manifest.Version.String(), string(c.env), conf.GetFloat64("trace.sample.ratio"))
+	ctx, cancel := context.WithTimeout(context.TODO(), conf.GetDuration("trace.host.timeout"))
+	defer cancel()
```
Comment (cnkarz) [inline connector.go:95]: have you tested this? this line should trigger cancel after returning the connector.

Comment (gunnarsundberg) [reply connector.go:95]: yeah afaik all it does is release child context from memory if it returns before the timeout is reached. correct me if I'm wrong

**Review by rukolahasser — APPROVED** [review]

**Pull request merged**